### PR TITLE
Cleanup the old backup infra resources

### PR DESCRIPTION
**Pull request by swapnilgm**
Signed-off-by: Swapnil Mhamane <swapnil.mhamane@sap.com>

**What this PR does / why we need it**:
This PR removes the orphan backupInfra resources which are no longer needed post reconciliation of Shoot using gardener 0.29+. 
**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```noteworthy operator
The orphaned `BackupInfrastructure` resources post reconciliation of shoots using Gardener 0.29+ are now getting deleted.
:warning: For hibernated cluster it now guarantees only latest snapshot will be there on shared bucket. It doesn't migrate the old snapshots and hence break the policy of the configured deletion grace period. If you want to keep the old data you would need to download it manually from the buckets before upgrading to this Gardener version.
```
```action operator
The old `garden.sapcloud.io/v1beta1.BackupInfrastructure` resources are deprecated and will be removed in the next release. The `core.gardener.cloud/v1alpha1.BackupBucket` and `core.gardener.cloud/v1alpha1` are the replacement.
```

### CODE REVIEW

**Comment by swapnilgm** [comment]
@rfranzke PR is ready for review.

File: pkg/operation/hybridbotanist/controlplane.go [hunk 1]
```diff
@@ -521,15 +520,26 @@ func (b *HybridBotanist) DeployETCD(ctx context.Context) error {
 		}
 
 		if b.Shoot.HibernationEnabled {
-			statefulset := &appsv1.StatefulSet{}
```
Comment (rfranzke) [inline pkg/operation/hybridbotanist/controlplane.go:524]: Why is this code removed? I think we should still have it, and move the whole `if lastSnapshotRevision > 0 && ...` part into the `if statefulset.Spec.Replicas == nil` section.
WDYT?

File: pkg/operation/botanist/waiter.go [hunk 2]
```diff
@@ -70,7 +71,7 @@ func (b *Botanist) WaitUntilEtcdReady(ctx context.Context) error {
 
 		bothEtcdStatefulSetsReady := true
 		for _, statefulSet := range statefulSetList.Items {
-			if statefulSet.DeletionTimestamp != nil {
+			if statefulSet.DeletionTimestamp != nil || statefulSet.Generation != statefulSet.Status.ObservedGeneration {
```
Comment (rfranzke) [inline pkg/operation/botanist/waiter.go]: Why was this added? If the generation is not equals to the observed generation then we should set `bothEtcdStatefulSetsReady=false`.

Comment (swapnilgm) [reply pkg/operation/botanist/waiter.go]: Ahh yeah . Not needed.

File: pkg/operation/hybridbotanist/controlplane.go (hunk 1 above)
Comment (swapnilgm) [reply pkg/operation/hybridbotanist/controlplane.go:524]: When we know replicas could be 1 or 0. I didn't get why did we had unnecessary `Get` call there.  So, removed it.

Comment (rfranzke) [reply pkg/operation/hybridbotanist/controlplane.go:524]: It's because of hibernation - when a shoot is hibernated the etcd is only scaled down at the very end of the flow.
Please revert the behaviour.

Comment (swapnilgm) [reply pkg/operation/hybridbotanist/controlplane.go:524]: Oh OK. I was unaware of it. For me change worked. So, didn't noticed it. On it.

**Comment by swapnilgm** [comment]
Thank you for the review. I have addressed the comments. PTAL.

**Review by rfranzke — APPROVED** [review]
